tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -14,7 +14,8 @@ include(`common/tlv.m4')
 include(`sof/tokens.m4')
 
 # Include ICL DSP configuration
-include(`platform/intel/icl.m4')
+ifelse(PLATFORM, `icl', include(`platform/intel/icl.m4'),
+	ifelse(PLATFORM, `cml', include(`platform/intel/cnl.m4'), `'))
```

Comment:
add a if else line 133 so that SDW2 is gone

Reply:
@plbossart I need help here. The ifelse doesn't work properly if PIPELINE_PCM_DAI_ADD or DAI_ADD is in the body.
See the patch [tplg.diff](https://github.com/thesofproject/sof/files/3762174/tplg.txt)
The reason is that there is a comma "," when PIPELINE_PCM_DAI_ADD or DAI_ADD translate its definition.
And the comma is the separator symbol of ifelse. So you will see below in the .conf file




```
SectionGraph."pipe-pass-playback-4" {
	index "4"

	lines [
		"BUF4.0
```
```
SectionGraph."ALH0x202" {
	index "4"

	lines [
		"ALH0x202.OUT
```
[sof-icl-rt711-rt1308-rt715-hdmi.conf](https://github.com/thesofproject/sof/files/3762131/sof-icl-rt711-rt1308-rt715-hdmi.conf.txt)

Reply:
what a stupid language ....

Reply:
@plbossart @bardliao m4 is not stupid at all. its quite elegant on the contrary.

What you are missing is a `' around your PIPELINE_PCM_DAI_ADD and DAI_ADD macro calls.

Reply:
> @plbossart @bardliao m4 is not stupid at all. its quite elegant on the contrary.
> 
> What you are missing is a `' around your PIPELINE_PCM_DAI_ADD and DAI_ADD macro calls.

How stupid I was. I tried **""** **``** **''** ，but never tried `'. Updating it now. Thanks again @ranj063